persistence-modules/java-jpa-3/src/test/java/com/baeldung/jpa/hibernateunproxy/HibernateProxyIntegrationTest.java

```diff
@@ -0,0 +1,75 @@
+package com.baeldung.jpa.hibernateunproxy;
+
+import org.hibernate.Hibernate;
+import org.hibernate.proxy.HibernateProxy;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import java.math.BigDecimal;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class HibernateProxyIntegrationTest {
+
+    private static EntityManager entityManager;
+
+    @BeforeAll
+    public static void setup() {
+        EntityManagerFactory factory = Persistence.createEntityManagerFactory("jpa-h2-hibernate-unproxy");
+        entityManager = factory.createEntityManager();
+        populateH2DB();
+    }
+
+    @Test
+    public void givenPaymentReceipt_whenAccessingPayment_thenVerifyType() {
+        PaymentReceipt paymentReceipt = entityManager.find(PaymentReceipt.class, 3L);
+        Assert.assertEquals(true, paymentReceipt.getPayment() instanceof HibernateProxy);
```

Comment:
Instead of Assert.assertEquals(true, expr), just use Assert.assertTrue(expr). Do this for all three examples that are using this method.

Reply:
Thanks! I've pushed the changes; also, marking this as resolved.